mmengine/hooks/sync_buffer_hook.py

```diff
@@ -13,6 +13,24 @@ class SyncBuffersHook(Hook):
 
     def __init__(self) -> None:
         self.distributed = is_distributed()
+        # A flag to mark whether synchronization has been done in
+        # after_train_epoch
+        self.called_in_train = False
+
+    def before_val_epoch(self, runner) -> None:
+        """All-reduce model buffers before each validation epoch.
+
+        Synchronize the buffers before each validation if they have not been
+        synchronized at the end of the previous training epoch. This method
+        will be called when using IterBasedTrainLoop.
+
+        Args:
+            runner (Runner): The runner of the training process.
+        """
+        if self.distributed:
+            if not self.called_in_train:
+                all_reduce_params(runner.model.buffers(), op='mean')
+            self.called_in_train = False
```

Comment:
But `all_reduce_params` will only be called when `self.called_in_train` is False. I think we need to always set `self.called_in_train` to False here. Otherwise `self.called_in_train` will always be True and never get updated.

Reply:
`self.called_in_train` is set as False in `__init__` so `all_reduce_params` will be called in `before_val_epoch` when using `IterBasedTrainLoop`. After `all_reduce_params` is called, the `self.called_in_train` will always be False.

Reply:
Yes, I agree. I believe both implementation will work correctly in `IterBasedTrainLoop`(and all other cases we have now). But could you please briefly explain why its necessary to only set `self.called_in_train` as False when `all_reduce_params`  is called. Is this for code clarity or something?  As I think always setting self.called_in_train to False may seem like a safer approach.

Reply:
It may look clear but your implementation is also correct so there is no need to update. Thanks for your contribution.